metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DepublishRecordIdDao.java

```diff
@@ -309,14 +315,17 @@ public long countSuccessfullyDepublishedRecordIdsForDataset(String datasetId) {
    * provided</p>
    *
    * @param datasetId the dataset for which to do this. Cannot be null
+   * @param recordIds the records for which to set this. Can be null or empty, in which case the
+   * operation will be performed on all records. If it is not empty, a new record will be created if
+   * a record with the given record ID is not already present.
    * @param depublicationStatus the depublication status. Cannot be null
    * @param depublicationDate the depublication date. Can be null only if depublicationStatus is
    * {@link DepublicationStatus#PENDING_DEPUBLICATION}
    */
-  public void markRecordIdsWithDepublicationStatus(String datasetId,
+  public void markRecordIdsWithDepublicationStatus(String datasetId, Set<String> recordIds,
```

Comment:
Why are we adding records here, I feel we shouldn't be doing that..?

Reply:
Imagine the following situation. 
1. Trigger depublication of records.
2. Remove those records from the list while the depublication is running
3. The depublication finishes, but it no longer has the record to set the DEPUBLISHED status.
I think we should reintroduce the record and set the status so that the user knows what happened.
(also, this is what the specs say - even though I wrote them myself :) )

Reply:
Perhaps we should not allow editing the list when an execution is running, instead?
Same thing we do with updating a workflow..

Reply:
Actually, it is possible to edit a workflow while an execution is in progress, isn't it?
But perhaps this is the solution in this case.
However, perhaps it would be best to do this in a separate ticket? I feel reluctant to change the scope of this ticket now, and whether we (I) change this now, or later, doesn't really matter. Also that gives us (me) some time to think about this. What do you think?

Reply:
As discussed we'll handle this as a separate ticket